core/mm3Types.js

```diff
@@ -65,10 +163,61 @@ CLASS({
       help: 'Regex pattern for property.'
     },
     {
+      name: 'minChars',
+      label: 'Minimum characters',
+      help: 'The minimum number of characters required.'
+    },
+    {
+      name: 'maxChars',
+      label: 'Maximum characters',
+      help: 'The maximum number of characters allowed.'
+    },
+    {
       name: 'prototag',
       label: 'Protobuf tag',
       required: false,
       help: 'The protobuf tag number for this field.'
+    },
+    {
+      name: 'validate',
+      lazyFactory: function() {
+        var prop = this; // this == the property
+        var ret = constantFn('');
+
+        var min = prop.minChars;
+        if ( typeof min !== 'string' || min ) { // ignore '' (falsey string)
```

Comment:
It expects either a number or a non empty string (we convert to a number).
Empty string is the "don't check me" case.

On Thu, Dec 10, 2015 at 10:41 AM adamvy <notifications@github.com> wrote:

> In core/mm3Types.js
> <https://github.com/foam-framework/foam/pull/491#discussion_r47241284>:
>
> >        name: 'prototag',
> >        label: 'Protobuf tag',
> >        required: false,
> >        help: 'The protobuf tag number for this field.'
> > +    },
> > +    {
> > +      name: 'validate',
> > +      lazyFactory: function() {
> > +        var prop = this; // this == the property
> > +        var ret = constantFn('');
> > +
> > +        var min = prop.minChars;
> > +        if ( typeof min !== 'string' || min ) { // ignore '' (falsey string)
>
> Why not just check typeof min == "number" ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/foam-framework/foam/pull/491/files#r47241284>.
>